File: frontend/lib/models/enum/role.dart
// ignore_for_file: constant_identifier_names

enum Role {
  admin('Admin', 'ADMIN'),
  viewer('Görüntüleyici', 'VIEWER'),
  fieldWorker('Saha Görevlisi', 'FIELD_WORKER'),
  fieldSupervisor('Saha Sorumlusu', 'FIELD_SUPERVISOR');

  final String displayName;
  final String apiValue;

  const Role(this.displayName, this.apiValue);

  static Role fromApiValue(String value) {
    return Role.values.firstWhere(
      (e) => e.apiValue == value,
      orElse: () => throw ArgumentError('Bilinmeyen Cisim Türü: $value'),
    );
  }

  bool get canDelete => this == Role.admin || this == Role.fieldSupervisor;
  bool get canChangeStatus =>
      this == Role.admin || this == Role.fieldSupervisor;
  bool get isAdmin => this == Role.admin;
  bool get canEditPhoto => this == Role.admin || this == Role.fieldWorker;
}

File: backend/cografi-nesne-takip/src/main/java/com/kutalmis/cografi_nesne_takip/entity/Role.java
package com.kutalmis.cografi_nesne_takip.entity;

public enum Role {
    ADMIN("Admin"),
    VIEWER("Görüntüleyici"),
    FIELD_WORKER("Saha Görevlisi"),
    FIELD_SUPERVISOR("Saha Sorumlusu");

    private final String displayName;

    Role(String displayName) {
        this.displayName = displayName;
    }

    public String getDisplayName() {
        return displayName;
    }

}